src/Action/Woopit/CreateDeliveryTrait.php

```diff
@@ -34,13 +50,30 @@ protected function createTask(array $data, string $type): Task
     {
         $location = $data['location'];
 
-        $streetAddress = sprintf('%s, %s',
-            implode(', ', array_filter([$location['addressLine1'], $location['addressLine2']])),
-            sprintf('%s %s', $location['postalCode'], $location['city'])
-        );
+        $streetAddress = sprintf('%s %s %s', $location['addressLine1'], $location['postalCode'], $location['city']);
 
         $address = $this->geocoder->geocode($streetAddress);
 
+        $streetDescription = null;
+
+        if (isset($location['addressLine2'])) {
+            $streetDescription = $location['addressLine2'];
+        }
+
+        if (isset($location['floor'])) {
+            $streetDescription .= ' | Floor: ' . $location['floor'];
+        }
+
+        if (isset($location['doorCode'])) {
+            $streetDescription .= ' | Door code: ' . $location['doorCode'];
+        }
+
+        if (isset($location['comment'])) {
+            $streetDescription .= ' | ' . $location['comment'];
+        }
+
+        $address->setDescription($streetDescription);
+
```

Comment:
The same code seems to be repeated both in CreateDeliveryTrait and UpdateDeliveryTrait. Can we factorize it?